subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/TransformArtifactOperation.java

```diff
@@ -29,31 +30,35 @@
 
 class TransformArtifactOperation implements RunnableBuildOperation {
     private static final Logger LOGGER = LoggerFactory.getLogger(TransformArtifactOperation.class);
-    private final ResolvableArtifact artifact;
+    private final ComponentArtifactIdentifier artifactId;
+    private final File file;
     private final ArtifactTransformer transform;
     private Throwable failure;
     private List<File> result;
 
-    TransformArtifactOperation(ResolvableArtifact artifact, ArtifactTransformer transform) {
-        this.artifact = artifact;
+    TransformArtifactOperation(ComponentArtifactIdentifier artifactId,  File file, ArtifactTransformer transform) {
+        this.artifactId = artifactId;
+        this.file = file;
         this.transform = transform;
     }
 
     @Override
     public void run(@Nullable BuildOperationContext context) {
         try {
             if (LOGGER.isInfoEnabled()) {
-                LOGGER.info("Executing transform {} on artifact {}", transform.getDisplayName(), artifact.getId().getDisplayName());
+                LOGGER.info("Executing transform {} on artifact {}", transform.getDisplayName(), artifactId.getDisplayName());
             }
-            result = transform.transform(artifact.getFile());
+            result = transform.transform(file);
         } catch (Throwable t) {
             failure = t;
         }
     }
 
     @Override
     public BuildOperationDescriptor.Builder description() {
-        return BuildOperationDescriptor.displayName("Apply " + transform.getDisplayName() + " to " + artifact);
+        return BuildOperationDescriptor.displayName("Transform " + artifactId.getDisplayName() + " with " + transform.getDisplayName())
+            .progressDisplayName("Transform " + transform.getDisplayName())
```

Comment:
I think we should add the `file` to the progress display name, too. If the file is not there, we only see something like `Transform MinifyTransform` which is not very helpful IMO.

Reply:
Added. I think the most important part is that we see _something_ happening. We can figure out the best way to show progress based on real-world experience.